internal/cortex/querier/queryrange/results_cache.go

```diff
@@ -357,6 +360,45 @@ func (s resultsCache) isAtModifierCachable(r Request, maxCacheTime int64) bool {
 	return atModCachable
 }
 
+var negativeOffset = errors.New("negative offset")
+
+func (s resultsCache) isOffsetCachable(r Request) bool {
+	query := r.GetQuery()
+	if !strings.Contains(query, "offset") {
+		return true
+	}
+	expr, err := parser.ParseExpr(query)
+	if err != nil {
+		level.Warn(s.logger).Log("msg", "failed to parse query, considering @ modifier as not cachable", "query", query, "err", err)
```

Comment:
Why it is related to `@ modifier`?

Reply:
Isnt "@" super cacheable?

Reply:
I think he just copied over https://github.com/thanos-io/thanos/blob/main/internal/cortex/querier/queryrange/results_cache.go#L326.

`@` cannot be cached if it is a very recent or future timestamp

Reply:
Ah for future timestamp it will return empty now and in future might return something, right?

Reply:
yea :), didn't check the comment.

> I think he just copied over https://github.com/thanos-io/thanos/blob/main/internal/cortex/querier/queryrange/results_cache.go#L326.